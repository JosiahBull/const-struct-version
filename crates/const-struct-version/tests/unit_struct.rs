use const_struct_version::StructVersion;

#[derive(StructVersion)]
struct UnitStruct;

#[derive(StructVersion)]
struct UnitStruct2;

#[test]
fn test_unit_structs_give_same_hash() {
    let version = <UnitStruct as StructVersion>::version();
    let version2 = <UnitStruct2 as StructVersion>::version();

    insta::assert_debug_snapshot!(version);

    // Ensure these are the same.
    assert_eq!(version, version2);
}
