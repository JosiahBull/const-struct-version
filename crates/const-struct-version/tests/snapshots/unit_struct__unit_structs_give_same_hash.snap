---
source: crates/const-struct-version/tests/unit_struct.rs
expression: version
---
"da39a3ee5e6b4b0d3255bfef95601890afd80709"
